test/utils/nrosched/nrosched.go

```diff
@@ -93,42 +93,29 @@ func CheckPODWasScheduledWith(k8sCli *kubernetes.Clientset, podNamespace, podNam
 	return checkPODEvents(k8sCli, podNamespace, podName, schedulerName, isScheduledWith)
 }
 
-func CheckNROSchedulerAvailable(cli client.Client, nroSchedName string) *nropv1alpha1.NUMAResourcesScheduler {
-	nroSchedObj := &nropv1alpha1.NUMAResourcesScheduler{}
-	Eventually(func() bool {
-		By(fmt.Sprintf("checking %q for the condition Available=true", nroSchedName))
-
-		err := cli.Get(context.TODO(), client.ObjectKey{Name: NROSchedObjectName}, nroSchedObj)
-		if err != nil {
-			klog.Warningf("failed to get the scheduler resource: %v", err)
-			return false
-		}
-
-		cond := status.FindCondition(nroSchedObj.Status.Conditions, status.ConditionAvailable)
-		if cond == nil {
-			klog.Warningf("missing conditions in %v", nroSchedObj)
-			return false
-		}
-
-		klog.Infof("condition: %v", cond)
-
-		return cond.Status == metav1.ConditionTrue
-	}, 5*time.Minute, 10*time.Second).Should(BeTrue(), "Scheduler condition did not become available")
-	return nroSchedObj
+func CheckNROSchedulerAvailable(cli client.Client) *nropv1alpha1.NUMAResourcesScheduler {
```

Comment:
what's the benefit of removing the explicit scheduler name?

Reply:
It seemed a bit counter intuitive to me that we are trying to obtain the scheduler object and subsequently the scheduler name (GetNROSchedulerName) by accessing the API but already have a scheduler Name.

Reply:
true, to some extent. But here we managed to mislead ourselves (!!!). `nroSchedName` is the name of the NUMAResourcesScheduler object, which includes the scheduler name. The name of the object and the name of the scheduler (the latter to be used in podSpec) can very well differ.

Reply:
Ah, okay. I understand. I will make the relevant changes in the next iteration.